components/ai_chat/core/browser/engine/engine_consumer.h

```diff
@@ -85,6 +87,8 @@ class EngineConsumer {
       const std::string& page_content,
       const ConversationHistory& conversation_history,
       const std::string& selected_language,
+      const std::vector<const Tool*> tools,
```

Comment:
This should be `const std::vector<base::WeakPtr<Tool>>&` so we can guarantee the pointer is valid when we reference it. (It will be set to to null when the object is destroyed)